packages/lib-user/src/components/shared/GroupForm/GroupForm.js

```diff
@@ -0,0 +1,177 @@
+import { Box, Button, Form, FormField, RadioButtonGroup, Select, TextInput, ThemeContext } from 'grommet'
+import { func, node, shape, string } from 'prop-types'
+import { useEffect, useState } from 'react'
+import styled from 'styled-components'
+
+import FieldLabel from './components/FieldLabel'
+import RadioInputLabel from './components/RadioInputLabel'
+import formTheme from './theme'
+
+const StyledButton = styled(Button)`
+  border-radius: 4px;
+`
+
+const PRIVATE_STATS_VISIBILITY = [
+  {
+    label: 'No, never show individual stats',
+    value: 'private_agg_only',
+  },
+  {
+    label: 'Yes, always show individual stats',
+    value: 'private_show_agg_and_ind',
+  }
+]
+
+const PUBLIC_STATS_VISIBILITY = [
+  {
+    label: 'No, never show individual stats',
+    value: 'public_agg_only',
+  },
+  {
+    label: 'Yes, show individual stats if member',
+    value: 'public_agg_show_ind_if_member',
+  },
+  {
+    label: 'Yes, always show individual stats',
+    value: 'public_show_all',
+  }
+]
+
+const DEFAULT_HANDLER = () => true
+
+const DEFAULT_VALUE = {
+  display_name: '',
+  visibility: 'Private',
+  stats_visibility: 'private_agg_only'
+}
+
+function GroupForm({
+  children,
+  defaultValue = DEFAULT_VALUE,
+  handleDelete = DEFAULT_HANDLER,
+  handleSubmit = DEFAULT_HANDLER
+}) {
+  const [value, setValue] = useState(defaultValue)
+  const statsVisibilityOptions = value.visibility === 'Private' ? PRIVATE_STATS_VISIBILITY : PUBLIC_STATS_VISIBILITY
+
+  useEffect(() => {
+    setValue(defaultValue)
+  }, [defaultValue])
+
+  return (
+    <ThemeContext.Extend value={formTheme}>
+      <Box>
+        <Form
+          onChange={(nextValue, { touched }) => {
+            if (nextValue.visibility !== value.visibility) {
+              const statsVisibility = nextValue.visibility === 'Private' ? 'private_agg_only' : 'public_agg_only'
+              nextValue.stats_visibility = statsVisibility
+            }
+            setValue(nextValue)
+          }}
+          onSubmit={handleSubmit}
+          value={value}
+        >
+          <FormField
+            label={<FieldLabel>Group Name</FieldLabel>}
+            help={defaultValue?.id ? null : 'By creating a new Group you will become the admin.'}
+            htmlFor='display_name'
+            name='display_name'
+            required
+            validate={[
+              (name) => {
+                if (name && name.length < 4) return 'must be > 3 characters'
+                return undefined
+              }
+            ]}
+            validateOn='blur'
+          >
+            <TextInput
+              id='display_name'
+              name='display_name'
+            />
+          </FormField>
+          <FormField
+            label={<FieldLabel>Public Visibility</FieldLabel>}
+            htmlFor='visibility'
```

Comment:
I'm not sure how to have a border around the other FormFields but not have a border around this FormField (per Figma design).

I did reduce the FormField and FieldLabel margins to tighten the general form spacing.

Reply:
You can put `</ThemeContext.Extend>` around individual FormField components you want it to apply to, and exclude the FormField that does not have a border in the Figma design.

Reply:
👍 updated accordingly, with a few other small adjustments